lib/index.js

```diff
@@ -1,14 +1,81 @@
 /*!
- * Copyright (c) 2023 Digital Bazaar, Inc. All rights reserved.
+ * Copyright (c) 2023-2024 Digital Bazaar, Inc. All rights reserved.
 */
+import * as EcdsaMultikey from '@digitalbazaar/ecdsa-multikey';
 import {canonize} from './canonize.js';
 import {createVerifier} from './createVerifier.js';
 import {name} from './name.js';
 import {requiredAlgorithm} from './requiredAlgorithm.js';
+import {sha} from './sha.js';
 
 export const cryptosuite = {
   canonize,
   createVerifier,
+  createVerifyData: _createVerifyData,
   name,
-  requiredAlgorithm,
+  requiredAlgorithm
 };
+
+async function _createVerifyData({
+  cryptosuite, document, proof, documentLoader, dataIntegrityProof,
+  verificationMethod
+} = {}) {
+  // determine digest algorithm from key algorithm
+  let keyAlgorithm;
+  if(verificationMethod) {
+    const key = await EcdsaMultikey.from(verificationMethod);
+    const verifier = key.verifier();
+    keyAlgorithm = verifier.algorithm;
+  } else if(dataIntegrityProof.signer?.algorithm) {
+    keyAlgorithm = dataIntegrityProof.signer.algorithm;
+  }
+  if(!keyAlgorithm) {
+    throw new Error(
+      'Either "verificationMethod" or "signer" with "algorithm" must be ' +
+      'passed to cryptosuite to determine hash algorithm.');
+  }
+  const algorithm = keyAlgorithm === 'P-256' ? 'SHA-256' : 'SHA-384';
```

Comment:
Does this (and the sha.js file) need to be more explicit about hashes it handles and otherwise fail?  If you had a P-521 verificationMethod, would the code get here with that keyAlgorithm and pick the wrong hash?